### PR TITLE
Updated README and all related documentation to community standards

### PR DESCRIPTION
**Pull request by rayrod2030**
This pull request updates the README and all other related documentation to chef cookbook community standards.

Summary of changes:

* Updated README with standard formatting
* Updated Copyright year in most places
* Added CONTRIBUTING and ISSUES documentation
* Moved attributes documentation from README over to attributes file
* Updated README to reflect testing on Ubuntu 14.04LTS.

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -1,40 +1,74 @@
...
+
+# Optional. Work around https://github.com/Netflix/ice/issues/100 by pre-creating
+# the directories it expects to find. Default: false
 node.default['ice']['processor']['issue_100_workaround']    = false
 
-# S3 keys for accessing billing files
+# AWS access key id used for accessing AWS detailed billing files from S3.
 node.default['ice']['billing_aws_access_key_id']            = nil
+# AWS secret key used for accessing AWS detailed billing files from S3.
 node.default['ice']['billing_aws_secret_key']               = nil
```
Comment (mjcdiggity) [inline attributes/default.rb]: These values are not seperated by a line, when most other values are. I know they are related, just stylistic choice that is up to you.

File: attributes/default.rb [hunk 2]
```diff
@@ -1,40 +1,74 @@
...
 
+# Name of the S3 bucket containing the AWS detailed billing files.
 node.default['ice']['billing_s3_bucket_name']               = nil
+# Directory in the S3 billing bucket containing AWS detailed billing files.
 node.default['ice']['billing_s3_bucket_prefix']             = nil
```
Comment (mjcdiggity) [inline attributes/default.rb]: Same line comment as above.

File: attributes/default.rb [hunk 3]
```diff
@@ -1,40 +1,74 @@
...
 
-node.default['ice']['billing_payerAccountId']				= nil
-node.default['ice']['billing_accessRoleName']				= nil
-node.default['ice']['billing_accessExternalId']				= nil
+node.default['ice']['billing_payerAccountId']               = nil
+node.default['ice']['billing_accessRoleName']               = nil
+node.default['ice']['billing_accessExternalId']             = nil
```
Comment (mjcdiggity) [inline attributes/default.rb]: No comment explaining values here, and no line seperation. While the names explain the purpose of the attribute, I think an explanation would be nice, and have the same style as other attributes.

File: attributes/default.rb [hunk 4]
```diff
@@ -1,40 +1,74 @@
...
 
+# Enables the Ice reader and installs the Nginx reverse proxy.
 node.default['ice']['reader']['enabled']                    = true
-node.default['ice']['reader']['local_dir']                   = '/var/ice_reader'
+# Local work directory for the Ice reader.
+node.default['ice']['reader']['local_dir']                  = '/var/ice_reader'
 
-node.default['ice']['start_millis']                          = 0
+# Specify the start time in milliseconds for the processor to start processing billing files.
+node.default['ice']['start_millis']                         = 0
```
Comment (mjcdiggity) [inline attributes/default.rb]: What time is this exactly? Date time, or time after startup? It is not clear.

File: attributes/default.rb [hunk 5]
```diff
@@ -1,40 +1,74 @@
...
 
+# Optional.  Hash mapping of AWS account names to account numbers.  This is used
+# within Ice to give accounts human readable names in the UI.
 node.default['ice']['accounts']                             = {}
 
+# Hash mapping accounts to IAM roles in order to allow reservations in that
+# account to be polled.
 node.default['ice']['reservation_capacity_poller']          = false
 
 node.default['ice']['iam_role']                             = 'ice'
```
Comment (mjcdiggity) [inline attributes/default.rb]: No comment about this iam role.

File: attributes/default.rb [hunk 6]
```diff
@@ -1,40 +1,74 @@
...
+# Specify the start time in milliseconds for the processor to start processing billing files.
+node.default['ice']['start_millis']                         = 0
 
+# Optional.  Hash mapping of AWS account names to account numbers.  This is used
+# within Ice to give accounts human readable names in the UI.
 node.default['ice']['accounts']                             = {}
 
+# Hash mapping accounts to IAM roles in order to allow reservations in that
+# account to be polled.
 node.default['ice']['reservation_capacity_poller']          = false
```
Comment (mjcdiggity) [inline attributes/default.rb]: Comment say hash map, but attribute appears to be a boolean value. If hash, should default to {}.

File: attributes/default.rb [hunk 7]
```diff
@@ -42,29 +76,71 @@
 # # reservation utilization, possible values are LIGHT, HEAVY
 node.default['ice']['reservation_utilization']               = 'HEAVY'
```
Comment (mjcdiggity) [inline attributes/default.rb]: Line comment same as above.

File: attributes/default.rb [hunk 8]
```diff
@@ -42,29 +76,71 @@
...
+
+# URL to highstock.js if you need to serve this over HTTPS (which the default
+# Highstock CDN does not currently support)
 node.default['ice']['highstockUrl']                          = nil
+
+# Monthly cache size.
 node.default['ice']['monthlycachesize']                      = nil
+
+# Cost per monitor metric per hour.
 node.default['ice']['cost_per_monitormetric_per_hour']       = nil
```
Comment (mjcdiggity) [inline attributes/default.rb]: Should monitor and metric be split with `_`?

File: attributes/default.rb [hunk 9]
```diff
@@ -42,29 +76,71 @@
...
 node.default['ice']['ondemandCostAlertEmails']               = nil
+
+# If set to `original`, Ice will use the original costs from the billing file
+# for Resource Groups.
 node.default['ice']['resourceGroupCost']                     = nil
+
+# Set to `true` to enable weekly cost emails.
 node.default['ice']['weeklyCostEmails']                      = nil
+
+# Email address from which weekly cost emails are sent (must be registered in SES)
```
Comment (mjcdiggity) [inline attributes/default.rb]: `.` at the end of this comment?

**Comment by mjcdiggity** [comment]
Overall looks good, just minor typos within attributes file. One other issue I found was the use of `camelCase` vs `snake_case` vs `camelSnake_case`. Which ever one is chosen, all attributes names should follow that pattern. In addition, some comments end with periods and some don't. Again whatever is chosen, all comments should follow. That is all.

File: attributes/default.rb (hunk 3 above)
Comment (rayrod2030) [inline attributes/default.rb]: Adding docs from netflix/ice.

File: attributes/default.rb (hunk 5 above)
Comment (rayrod2030) [inline attributes/default.rb]: Good catch here.  I believe this attribute which is passed to JAVA_OPTIONS is no longer recognized by Ice so removing from attributes and default recipe.

File: attributes/default.rb (hunk 6 above)
Comment (rayrod2030) [inline attributes/default.rb]: yup comments are in the wrong spot there.

File: attributes/default.rb (hunk 8 above)
Comment (rayrod2030) [inline attributes/default.rb]: This was submitted by another contributor.  Will fix and rerun tests.

**Comment by rayrod2030** [comment]
Fixing camelcase/snakecase stuff as well.  Stay tuned.

**Comment by rayrod2030** [comment]
On second thought changing too many attributes could break some configs.  Perhaps wait for those changes for a future PR?

**Comment by mjcdiggity** [comment]
@rayrod2030 agreed if these are updates for version 0.2.13, just make comment changes for now then, and make attribute changes in future PR, which will involve minor version bump.

File: attributes/default.rb (hunk 4 above)
Comment (rayrod2030) [inline attributes/default.rb]: Explaining that start_millis is milliseconds since start of unix epoch time.  Setting default value to 90 days in the past.

**Comment by mjcdiggity** [comment]
:boom:

**Pull request merged**